quetz/tests/test_config.py

```diff
@@ -83,56 +86,61 @@ def test_config_with_path(config_dir, config_base):
     c_one = Config(one_path)
 
     assert c_one.configured_section("users")
-    assert c_one.users_admins == ["one"]
+    assert c_one.users.admins == ["one"]
 
     c_other = Config(other_path)
 
     assert c_other.configured_section("users")
-    assert c_other.users_admins == ["other"]
+    assert c_other.users.admins == ["other"]
 
     c_new = Config(one_path)
 
     assert c_new is c_one
 
 
-def test_config_extend_require(config):
-    with pytest.raises(ConfigError):
-        config.register(
-            [
-                ConfigSection(
-                    "other_plugin",
-                    [
-                        ConfigEntry("some_config_value", str),
-                    ],
-                )
-            ]
-        )
-    # remove last entry again
-    config._config_map.pop()
+@pytest.fixture
+def _setup_config_extend_require():
+    from pydantic import BaseSettings
+
+    class OtherPluginSetting(BaseSettings):
+        some_config_value: str
+
+    Config.register(other_plugin=(OtherPluginSetting, ...))
+    yield
+    del Settings.__fields__["other_plugin"]
+    del Settings.__annotations__["other_plugin"]
```

Comment:
Do you need to explicitly do cleanup?

Can't you control this via the scope of the fixture?

Reply:
Might be difficult, since we're changing a class, not an instance of a class.